File: core/src/main/java/com/triangleleft/flashcards/ui/main/IMainView.java
package com.triangleleft.flashcards.ui.main;

import com.annimon.stream.Optional;
import com.triangleleft.flashcards.service.vocabular.VocabularyWord;
import com.triangleleft.flashcards.ui.common.view.IView;

public interface IMainView extends IView {

    void setTitle(String title);

    void showWord(Optional<VocabularyWord> word);

    void showList();

    void finish();

    void reloadList();

}

File: core/src/main/kotlin/com/triangleleft/flashcards/ui/main/IMainView.kt
package com.triangleleft.flashcards.ui.main

import com.triangleleft.flashcards.service.vocabular.VocabularyWord
import com.triangleleft.flashcards.ui.common.view.IView
import io.reactivex.Observable

interface IMainView : IView {

    fun render(it: MainViewState)

    fun backPresses(): Observable<Any>

    fun wordClicks(): Observable<VocabularyWord>

}
